Framework/DataHandling/src/LoadNXcanSAS.cpp

```diff
@@ -568,5 +549,53 @@ void LoadNXcanSAS::exec() {
   setProperty("OutputWorkspace", ws);
 }
 
+void LoadNXcanSAS::loadTransmission(H5::Group &entry, const std::string &name) {
+  if (!hasTransmissionEntry(entry, name)) {
+    g_log.information("NXcanSAS file does not contain transmission for " +
+                      name);
+    return;
+  }
+
+  auto transmission =
+      entry.openGroup(sasTransmissionSpectrumGroupName + "_" + name);
+
+  // Create a 1D workspace
+  auto tDataSet = transmission.openDataSet(sasTransmissionSpectrumT);
+  auto workspace = createWorkspaceForHistogram(tDataSet);
+
+  // Load logs
+  loadLogs(entry, workspace);
+  auto title = workspace->getTitle();
+  const std::string transExtension = "_trans_" + name;
+  title += transExtension;
+  workspace->setTitle(transExtension);
+
+  // Load instrument
+  loadInstrument(entry, workspace);
+
+  // Load transmission data
+  loadTransmissionData(transmission, workspace);
+
+  // Set to distribution
+  workspace->setDistribution(true);
+  workspace->setYUnitLabel("Transmission");
+
+  // Set the workspace on the output
+
+  std::string propertyName;
+  if (name == "sample")
+    propertyName = "TransmissionWorkspace";
+  else
+    propertyName = "TransmissionCanWorkspace";
+  const std::string doc = "The transmission workspace";
+
+  declareProperty(
```

Comment:
In which case there is a discrepancy between [`LoadCanSAS1D`](http://docs.mantidproject.org/nightly/algorithms/LoadCanSAS1D-v2.html) and [`LoadNXcanSAS`](http://docs.mantidproject.org/v3.8.0/algorithms/LoadNXcanSAS-v1.html). If we expose the properties in the algorthim dialog, then they would only provide a result if they are queried, right? Unfortunately the users expect them to appear without having to provide a name for them (the name is constructed from some file information). Is there a way around this?